plugin/src/main/java/com/denizenscript/denizen/scripts/commands/entity/FakeInternalDataCommand.java

```diff
@@ -0,0 +1,96 @@
+    // @Group entity
+    //
+    // @Description
+    // Sends fake internal entity data updates, optionally sending multiple over time.
+    // This supports sub-tick precision, allowing smooth/high FPS animations.
+    //
+    // The input to 'data:' is a list of <@link object MapTag>s, with each map being a frame to send, with each map being formatted like <@link mechanism EntityTag.internal_data>'s input.
+    //
+    // Optionally specify a list of players to fake the data for, defaults to the linked player.
+    //
+    // 'speed:' is the amount of time between each frame getting sent, supporting sub-tick delays.
+    //
+    // @Usage
+    // Animates an item display entity's item for the linked player, and slowly scales it up.
+    // - fakeinternaldata entity:<[item_display]> data:[item=iron_ingot;scale=0.6,0.6,0.6]|[item=gold_ingot;scale=0.8,0.8,0.8]|[item=netherite_ingot;scale=1,1,1] speed:0.5s
+    //
+    // @Usage
+    // Animates a rainbow glow on a display entity for all online players.
+    // - define color <color[red]>
+    // - repeat 361 from:0 as:hue:
+    //   - define frames:->:<map[glow_color=<[color].with_hue[<[hue]>].argb_integer>]>
+    // - fakeinternaldata entity:<[display]> data:<[frames]> for:<server.online_players> speed:0.01s
+    // -->
+
+    public static void autoExecute(ScriptEntry scriptEntry,
+                                   @ArgName("entity") @ArgPrefixed EntityTag inputEntity,
+                                   @ArgName("data") @ArgPrefixed @ArgSubType(MapTag.class) List<MapTag> data,
+                                   @ArgName("for") @ArgPrefixed @ArgDefaultNull @ArgSubType(PlayerTag.class) List<PlayerTag> forPlayers,
+                                   @ArgName("speed") @ArgPrefixed @ArgDefaultText("0s") DurationTag speed) {
+        List<Player> sendTo;
+        if (forPlayers != null) {
+            sendTo = new ArrayList<>(forPlayers.size());
+            for (PlayerTag player : forPlayers) {
+                sendTo.add(player.getPlayerEntity());
+            }
+        }
+        else if (Utilities.entryHasPlayer(scriptEntry)) {
+            sendTo = List.of(Utilities.getEntryPlayer(scriptEntry).getPlayerEntity());
+        }
+        else {
+            throw new InvalidArgumentsRuntimeException("Must specify players to fake the internal data for.");
+        }
+        Entity entity = inputEntity.getBukkitEntity();
+        List<List<Object>> frames = new ArrayList<>(data.size());
+        for (MapTag frame : data) {
+            frames.add(NMSHandler.entityHelper.convertInternalEntityDataValues(entity, frame));
+        }
+        long ms = speed.getMillis();
+        DenizenCore.runAsync(() -> {
+            try {
+                for (List<Object> frame : frames) {
+                    NMSHandler.packetHelper.sendEntityDataPacket(sendTo, entity, frame);
+                    Thread.sleep(ms);
```

Comment:
Should either note in meta that the delay is inaccurate for long runs, or switch this sleep to a more accurate delay manager.

Also considering the possibility of long runs, should also add a check to kill the loop if the player(s) are gone

Reply:
Both should be fixed